tests/hardhat/Fork/supply.ts

```diff
@@ -0,0 +1,325 @@
+    .connect(impersonatedTimelock)
+    .giveCallPermission(comptroller.address, "setCollateralFactor(address,uint256,uint256)", ADMIN);
+}
+
+if (FORK_TESTNET) {
+  describe("Supply fork tests", async () => {
+    async function setup() {
+      await setForkBlock(30080357);
+      await configureTimelock();
+
+      acc1Signer = await initMainnetUser(acc1, ethers.utils.parseUnits("2"));
+      acc2Signer = await initMainnetUser(acc2, ethers.utils.parseUnits("2"));
+      acc3Signer = await initMainnetUser(acc3, ethers.utils.parseUnits("2"));
+
+      usdt = FaucetToken__factory.connect(USDT, impersonatedTimelock);
+      usdd = MockToken__factory.connect(USDD, impersonatedTimelock);
+      vUSDT = await configureVToken(VUSDT);
+      vUSDD = await configureVToken(VUSDD);
+      comptroller = Comptroller__factory.connect(COMPTROLLER, impersonatedTimelock);
+      priceOracle = ChainlinkOracle__factory.connect(ORACLE, impersonatedOracleOwner);
+
+      await grantPermissions();
+
+      await comptroller.setMarketSupplyCaps(
+        [vUSDT.address, vUSDD.address],
+        [convertToUnit(1, 50), convertToUnit(1, 50)],
+      );
+      await comptroller.setMarketBorrowCaps(
+        [vUSDT.address, vUSDD.address],
+        [convertToUnit(1, 50), convertToUnit(1, 50)],
+      );
+      await comptroller.connect(acc1Signer).enterMarkets([vUSDT.address]);
+      await comptroller.connect(acc2Signer).enterMarkets([vUSDT.address, vUSDD.address]);
+      await comptroller.connect(acc3Signer).enterMarkets([vUSDD.address]);
+    }
+
+    beforeEach(async () => {
+      await setup();
+      await priceOracle.setDirectPrice(usdt.address, convertToUnit("1", 15));
+    });
+
+    const calculateExchangeRate = async () => {
+      const cash = await vUSDT.getCash();
+      const borrows = await vUSDT.totalBorrows();
+      const badDebt = await vUSDT.badDebt();
+      const reserves = await vUSDT.totalReserves();
+      const supply = await vUSDT.totalSupply();
+      const exchangeRatecal = new BigNumber(Number(cash) + Number(borrows) + Number(badDebt) - Number(reserves))
+        .multipliedBy(Number(convertToUnit(1, 18)))
+        .toFixed(0);
+
+      return new BigNumber(exchangeRatecal).dividedBy(Number(supply)).toFixed(0);
+    };
+
+    it("Evolution of exchange rate", async () => {
+      const mintAmount = convertToUnit("1", 17);
+      // Accural all the interest till latest block
+      await vUSDT.accrueInterest();
+
+      // Assert current exchange rate
+      let exchangeRate = await vUSDT.callStatic.exchangeRateCurrent();
+      let calculatedRate = await calculateExchangeRate();
+      expect(exchangeRate).closeTo(calculatedRate, 1);
+
+      // Mint vUSDT with first account(acc1)
+      await usdt.connect(acc1Signer).allocateTo(acc1, convertToUnit(2, 18));
+      await usdt.connect(acc1Signer).approve(vUSDT.address, convertToUnit(2, 18));
+      await expect(vUSDT.connect(acc1Signer).mint(convertToUnit(1, 18))).to.emit(vUSDT, "Mint");
```

Comment:
Any special reason to prepare 2 USDT and then only supply 1 USDT?

Reply:
Yes, funds were used during liquidation.